### ISSUE TITLE
ETL merge transformer doesn't allow to update a vertex anymore

### ISSUE THREAD

**Issue by padutilleul** [orientechnologies/orientdb#5346]
Hello,

Until orientDB 2.1.0 (at least), you could use the merge transfomer in the ETL module to update an existing vertex using an unique identifier.

This had multiple uses, including : 
- allowing to import a tree in any order, by creating children vertexes first and using the ``"unresolvedLinkAction": "CREATE"`` option in the Edge transformer to create the parents, then updating them
- updating an existing graph from a data source

However, in 2.1.5, it doesn't seem to work anymore

Here is a simple ETL json config that allow to recreate this problem : 
```
{
  "config": {
    "log": "DEBUG"
  },
  "source": {
    "content": { "value": "name,idf\nnode,1" }
  },
  "extractor" : {
    "row": {}
  },
  "transformers" : [
    {"csv": {}},
    {"merge": { "joinFieldName":"idf", "lookup":"V.idf"}},
    { "vertex": { "class": "V"} }
  ],
  "loader" : {
    "orientdb": {
      "dbURL": "remote:localhost/Bug",
      "dbUser": "root",
      "dbPassword": "***",
      "dbType": "graph",
      "indexes": [
        {"class":"V", "fields":["idf:integer"], "type":"UNIQUE" }
      ]
    }
  }
}
```

If you run it once, it works, but running it a second time gets you this error : 
```
OrientDB etl v.2.1.5 (build 2.1.x@r; 2015-10-29 16:54:25+0000) www.orientdb.com
BEGIN ETL PROCESSOR
Error in Pipeline execution: com.orientechnologies.orient.core.storage.ORecordDuplicatedException: Cannot index record V{name:node1,idf:1}: found duplicated key '1' in index 'V.idf' previously assigned to the record #9:0 RID=#9:0
ETL process halted: com.orientechnologies.orient.etl.OETLProcessHaltedException: Halt
Exception in thread "main" com.orientechnologies.orient.etl.OETLProcessHaltedException: Halt
	at com.orientechnologies.orient.etl.OETLPipeline.execute(OETLPipeline.java:149)
	at com.orientechnologies.orient.etl.OETLProcessor.executeSequentially(OETLProcessor.java:448)
	at com.orientechnologies.orient.etl.OETLProcessor.execute(OETLProcessor.java:255)
	at com.orientechnologies.orient.etl.OETLProcessor.main(OETLProcessor.java:109)
Caused by: com.orientechnologies.orient.core.storage.ORecordDuplicatedException: Cannot index record V{name:node1,idf:1}: found duplicated key '1' in index 'V.idf' previously assigned to the record #9:0 RID=#9:0
	at com.orientechnologies.orient.core.index.OIndexTxAwareOneValue.checkEntry(OIndexTxAwareOneValue.java:224)
	at com.orientechnologies.orient.core.index.OClassIndexManager.checkIndexedPropertiesOnCreation(OClassIndexManager.java:341)
	at com.orientechnologies.orient.core.index.OClassIndexManager.checkIndexes(OClassIndexManager.java:585)
	at com.orientechnologies.orient.core.index.OClassIndexManager.onRecordBeforeCreate(OClassIndexManager.java:410)
	at com.orientechnologies.orient.core.hook.ODocumentHookAbstract.onTrigger(ODocumentHookAbstract.java:226)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.callbackHooks(ODatabaseDocumentTx.java:1037)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.executeSaveRecord(ODatabaseDocumentTx.java:1930)
	at com.orientechnologies.orient.core.tx.OTransactionNoTx.saveRecord(OTransactionNoTx.java:160)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.save(ODatabaseDocumentTx.java:2532)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.save(ODatabaseDocumentTx.java:2373)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.save(ODatabaseDocumentTx.java:121)
	at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.createRecord(OBinaryNetworkProtocolAbstract.java:365)
	at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.createRecord(ONetworkProtocolBinary.java:1565)
	at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:361)
	at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:223)
	at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:77)
```

But you could run it multiple times without problem in 2.1.0

Is it a bug, or is there a new way to do this ?

**Comment by apiy72** [comment]
I am also getting the same issue #5341.

**Comment by robfrank** [comment]
it should works now. fixed on next release (2.1.7)
